mkosi/__init__.py

```diff
@@ -7275,6 +7279,21 @@ def setup_netdev(args: MkosiArgs, root: Path, do_run_build_script: bool, cached:
         run(["systemctl", "--root", root, "enable", "systemd-networkd"])
 
 
+def run_kernel_install(args: MkosiArgs, root: Path, do_run_build_script: bool, for_cache: bool) -> None:
+    if not args.bootable or do_run_build_script or for_cache:
+        return
+
+    with complete_step("Generating initramfs images…"):
+        # Running kernel-install on Debian/Ubuntu doesn't regenerate the initramfs. Instead, we can trigger
+        # regeneration of the initramfs via "dpkg-reconfigure dracut". kernel-install can then be called to put
+        # the generated initrds in the right place.
+        if args.distribution in (Distribution.debian, Distribution.ubuntu):
+            run_workspace_command(args, root, ["dpkg-reconfigure", "dracut"])
+
+        for kver, kimg in gen_kernel_images(args, root):
+            run_workspace_command(args, root, ["kernel-install", "add", kver, Path("/") / kimg])
```

Comment:
> `Path("/") / kimg`

That's ugly, and all the callers do this fixup. Please just make the path absolute in `gen_kernel_images()`.

Reply:
Frankly, I don't immediately see a cleaner way to do this since we need to support referring to the kernel image in the context of the image rootfs and in the context of the host rootfs. There's 3 options:

1. Return the path relative to the image rootfs. This is what we do now. It means we have to prefix with the "anchor" whenever trying to use the path, either `/` or `root`.
2. Return the absolute path in the image rootfs. This means we only have to prefix the path when using it in the context of the host rootfs. However, joining 2 absolute paths in python is not exactly intuitive

```
>>> from pathlib import Path
>>> a = Path("/usr")
>>> b = Path("/abc")
>>> a / b
PosixPath('/abc')
```

3. Return the absolute path in the host rootfs. This means we have to transform the path when trying to use it in the image rootfs context. This comes down to something like the following:

```
Path("/") / kimg.relative_to(root)
```

Both 2 and 3 don't seem super intuitive to me, so it seems better to have `gen_kernel_images()` return a path relative to the image rootfs and either prepend `/` or `root` as needed depending on the context.

Reply:
OK.